gensim/test/test_phrases.py

```diff
@@ -75,6 +84,20 @@ def testBigramConstruction(self):
         self.assertTrue(u'graph_minors' in self.bigram[sentences[-1]])
         self.assertTrue(u'human_interface' in self.bigram[sentences[-1]])
 
+    def testBigramConstructionFromGenerator(self):
+        """Test Phrases bigram construction building when corpus is a generator"""
+        bigram1_seen = False
+        bigram2_seen = False
+
+        for s in self.bigram[gen_sentences()]:
+            if not bigram1_seen and 'response_time' in s:
```

Comment:
This is a cop-out, but I just copied the format of the [existing test](https://github.com/RaRe-Technologies/gensim/blob/develop/gensim/test/test_phrases.py#L54) :P

To stream-line that section you could do:
```python
assert len(set(['response_time', 'graph_minors']).intersection(set(it.chain.from_iterable(self.bigram[gen_sentences()])))) == 2
```

However, the current formulation in both my test and [testBigramConsturction](https://github.com/RaRe-Technologies/gensim/blob/develop/gensim/test/test_phrases.py#L54) short-circuits so that it doesn't have to go through the entire input.  This would allow longer test corpora in the future.  

In either case, I didn't want to question or rethink the work of whoever designed the tests originally, I just wanted to make sure my changes were non-breaking.  Do you want me to change the tests?